playbook/app/pb_kits/playbook/pb_dialog/dialogHelper.js

```diff
@@ -32,6 +18,20 @@ const dialogHelper = () => {
       document.getElementById(closeTriggerData).close();
     });
   });
+
+  // Close dialog box on outside click
+  dialogs.forEach((dialogElement) => {
+    dialogElement.addEventListener("click", (event) => {
+      const dialogParentDataset = dialogElement.parentElement.dataset
+      if (dialogParentDataset.overlayClick === "overlay_close") return
+
+      const clickedOutsideDialogBox = event.target.classList.contains("pb_dialog_rails")
```

Comment:
@markdoeswork would we ever have a case where 2 dialogs would be open at once? If not, this is fine.

Reply:
Yes, Just as the sales mentor form has a specific dialog for "blank mentor" I would expect nitro devs to want dialogs for multiple reasons.

Or use cases where they're using some number of the 6 different status alert props

Reply:
@markdoeswork yeah that's fine but would they be open at the exact same time? If so, then on any click outside, both would close. In my mind, we would never have a case where two or more would be open at once right?

Reply:
@gavhuang That might be a little more of a product question for @co1in-1ewis I'd say...
I think there could be places where we want two to stay open. _But_ i'm also fine with making that harder to do... because it feels like a very rare case. @nidaqg @co1in-1ewis and I just looked at something similar for popover with filter. But let's check in with Colin and see.

Reply:
It may have changed, but there were some places in Nitro where the code was set up so that a dialog was nested within a dialog, so a full height dialog with another dialog trigger inside it. Would this refactor effect that sort of usecase @gavhuang ?

Reply:
@nidaqg good point! To address those situations (assuming this is an edge case), we could prevent both dialogs from closing on outside click with the prop.

Reply:
![Screenshot 2023-07-18 at 1 37 16 PM](https://github.com/powerhome/playbook/assets/9158723/dc6cc9ca-d3c5-4562-a87c-26f9146926af)
Yes. I think it's good. I would just resolve the weird spacing on that button first. @gavhuang

Reply:
@jasperfurniss this was only a local test of multiple dialog boxes. So not included in this PR :)